helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl

```diff
@@ -616,13 +620,13 @@ exporters:
     disable_compression: true
   {{- end }}
   {{- if (eq (include "splunk-otel-collector.platformLogsEnabled" .) "true") }}
-  {{- include "splunk-otel-collector.splunkPlatformLogsExporter" . | nindent 2 }}
+  {{- include "splunk-otel-collector.addPersistentStorage" (dict "exporter" (include "splunk-otel-collector.splunkPlatformLogsExporter" .) "isEnabled" (include "splunk-otel-collector.persistentQueueEnabled" .)) | nindent 2 }}
   {{- end }}
   {{- if (eq (include "splunk-otel-collector.platformMetricsEnabled" .) "true") }}
-  {{- include "splunk-otel-collector.splunkPlatformMetricsExporter" . | nindent 2 }}
+  {{- include "splunk-otel-collector.addPersistentStorage" (dict "exporter" (include "splunk-otel-collector.splunkPlatformMetricsExporter" .) "isEnabled" (include "splunk-otel-collector.persistentQueueEnabled" .)) | nindent 2 }}
   {{- end }}
   {{- if (eq (include "splunk-otel-collector.platformTracesEnabled" .) "true") }}
-  {{- include "splunk-otel-collector.splunkPlatformTracesExporter" . | nindent 2 }}
+  {{- include "splunk-otel-collector.addPersistentStorage" (dict "exporter" (include "splunk-otel-collector.splunkPlatformTracesExporter" .) "isEnabled" (include "splunk-otel-collector.persistentQueueEnabled" .)) | nindent 2 }}
```

Comment:
Can you modify `splunk-otel-collector.splunkPlatformTracesExporter` and other helpers instead of introducing `splunk-otel-collector.addPersistentStorage` helper? This is a scenario when following the DRY principle makes the code too hard to read.

Reply:
Well, here's a catch.
As it's only used in agent, modifying `splunk-otel-collector.splunkPlatformTracesExporter` directly without any additional data would be not possible. We use the same helper for the gateway.
We don't know which pipeline (agent or gateway) called `splunk-otel-collector.splunkPlatformTracesExporter`. 
I will try to implement in a much cleaner way.

Reply:
I did try something cleaner, https://github.com/signalfx/splunk-otel-collector-chart/pull/861/files#diff-55f9d185ea0bc213a43cf77ba9ebcba2f7ffd650e9535fba0bae53706fd45825R624.
I set a variable in the current scope before calling `splunkPlatformTracesExporter` and family. That way those helpers would know whether to include the `storage` or not. 
Let me know if that's fine by you.

Reply:
For clusterReceiver and gateway, we don't set that field. 
We only set it for the agent and unset it once we're done. 
Kind of weird but does the work.

Reply:
I find it much cleaner than passing extra variables to `include splunk-otel-collector.splunkPlatformTracesExporter` for all three exporters.

Reply:
This looks good to me. Thank you!